serial/serial_resources/component_instance.py

```diff
@@ -0,0 +1,112 @@
+import serial
+from ..api_client import APIClient
+from .component_instance_link import ComponentInstanceLink
+
+class ComponentInstance:
+    def __init__(self, component_instance_data):
+        self.client = APIClient(serial.api_key, serial.base_url)
+        self.data = component_instance_data
+        self.links = [] 
+
+    def add_link(self, link_name, child_identifier, break_prior_links=False, process_entry=None):
+        """
+        Creates a link between this component instance and a child at a specific
+        process_entry
+
+        Args: 
+        - link_name: User facing name for the link
+        - child_identifier: Identifier for the child component instance to be linked
+        - break_prior_links: Boolean for whether to break prior links
+        - process_entry?: Override for the process entry this component instance object 
+        was created with. Must be included if this component instance object was not 
+        created with a process entry.
+
+        Returns:
+        - New component instance link
+        """
+
+        if serial.debug:
+            print(f"Adding link: {link_name} with child identifier: {child_identifier}")
+        child_component_instance_params = {
+                "identifier": child_identifier
+                }
+        link_params = {
+                "name": link_name
+                }
+        child_component_instance = self.client.make_api_request("/components/instances",
```

Comment:
This always getting the [0]'th element of a list makes me feel a little awkward, but I did design the API like that. Wondering what you might think of it.

Reply:
I don't think it's that big of a deal. It's the same with supabase, which is why there's the .single() modifier and the supabaseSingleOverride. I've grown to dislike the supabaseSingleOverride function because it's not super readable. I think always indexing at 0 is fine. 

What happens if that child_identifier doesn't exist?

Reply:
Do we error at `child_component_instance["id"]`?

Reply:
We error before it - the api_client raises exceptions on non 2XX status codes